sql/catalog/lease: TestLeaseObserver failed

sql/catalog/lease.TestLeaseObserver [failed](https://mesolite.cluster.engflow.com/invocations/default/45ea8a30-8700-442d-a25a-1c3298c3febc?testReportRun=1&testReportShard=3&testReportAttempt=1#targets-Ly9wa2cvc3FsL2NhdGFsb2cvbGVhc2U6bGVhc2VfdGVzdA==) on master @ [e65e5180434403ffdc9683ad402c144447b2ea7e](https://github.com/cockroachdb/cockroach/commits/e65e5180434403ffdc9683ad402c144447b2ea7e):


```
I260410 20:58:29.323108 112450 sql/schemachanger/scrun/scrun.go:331  [T1,Vsystem,n1,job=NEW SCHEMA CHANGE id=1165834024648736769] 330  rebuilding declarative schema change state from descriptors [104]
I260410 20:58:29.323250 112450 sql/schemachanger/scplan/plan.go:97  [T1,Vsystem,n1,job=NEW SCHEMA CHANGE id=1165834024648736769] 331  building declarative schema changer plan in PostCommitPhase (rollback=false) for ALTER TABLE
I260410 20:58:29.324715 112450 sql/schemachanger/scrun/scrun.go:186  [T1,Vsystem,n1,job=NEW SCHEMA CHANGE id=1165834024648736769] 332  executing declarative schema change PostCommitPhase stage 1 of 1 with 3 MutationType ops (rollback=false) for ALTER TABLE
I260410 20:58:29.340768 112450 sql/internal.go:1995  [T1,Vsystem,n1,job=NEW SCHEMA CHANGE id=1165834024648736769] 333  waiting for 1 modified and 0 deleted descriptors
I260410 20:58:29.342358 112510 sql/catalog/lease/descriptor_state.go:198  [n1] 334  new lease: tabledesc.immutable: {ID: 104, Version: 3, ModificationTime: "1775854709.324637859,0", ParentID: 100, ParentSchemaID: 101, State: PUBLIC, NextColumnID: 3, Columns: [{ID: 1, TypeID: 20, Null: false}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2]}], Mutations: [{MutationID: 1, Direction: ADD, State: WRITE_ONLY, Column: {ID: 2, TypeID: 20, Null: true, State: ADD, MutationID: 1}}], PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}], StoreColumns: [2]}]}
I260410 20:58:29.343493 112450 sql/catalog/lease/lease.go:619  [T1,Vsystem,n1,job=NEW SCHEMA CHANGE id=1165834024648736769] 335  waiting for 1 leases to expire: desc=[{t 104 2}]
I260410 20:58:29.347431 112450 sql/catalog/lease/lease.go:613  [T1,Vsystem,n1,job=NEW SCHEMA CHANGE id=1165834024648736769] 336  all leases have expired at 1775854709.345972873,0: desc=[{t 104 2}]
I260410 20:58:29.347477 112450 sql/internal.go:2025  [T1,Vsystem,n1,job=NEW SCHEMA CHANGE id=1165834024648736769] 337  waiting for descriptors done, took 6.711504ms
I260410 20:58:29.347579 112450 sql/schemachanger/scrun/scrun.go:186  [T1,Vsystem,n1,job=NEW SCHEMA CHANGE id=1165834024648736769] 338  executing declarative schema change PostCommitNonRevertiblePhase stage 1 of 1 with 4 MutationType ops (rollback=false) for ALTER TABLE
I260410 20:58:29.363487 112450 5@util/log/event_log.go:90  [T1,Vsystem,n1,job=NEW SCHEMA CHANGE id=1165834024648736769] 339 ={"Timestamp":1775854709347502271,"EventType":"finish_schema_change","InstanceID":1,"DescriptorID":104,"LatencyNanos":55078058}
I260410 20:58:29.363540 112450 sql/internal.go:1995  [T1,Vsystem,n1,job=NEW SCHEMA CHANGE id=1165834024648736769] 340  waiting for 1 modified and 0 deleted descriptors
I260410 20:58:29.365491 112541 sql/catalog/lease/descriptor_state.go:198  [n1] 341  new lease: tabledesc.immutable: {ID: 104, Version: 4, ModificationTime: "1775854709.347502271,0", ParentID: 100, ParentSchemaID: 101, State: PUBLIC, NextColumnID: 3, Columns: [{ID: 1, TypeID: 20, Null: false}, {ID: 2, TypeID: 20, Null: true}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2]}], PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}], StoreColumns: [2]}]}
I260410 20:58:29.366104 112450 sql/catalog/lease/lease.go:619  [T1,Vsystem,n1,job=NEW SCHEMA CHANGE id=1165834024648736769] 342  waiting for 1 leases to expire: desc=[{t 104 3}]
I260410 20:58:29.369130 112450 sql/catalog/lease/lease.go:613  [T1,Vsystem,n1,job=NEW SCHEMA CHANGE id=1165834024648736769] 343  all leases have expired at 1775854709.367816878,0: desc=[{t 104 3}]
I260410 20:58:29.369176 112450 sql/internal.go:2025  [T1,Vsystem,n1,job=NEW SCHEMA CHANGE id=1165834024648736769] 344  waiting for descriptors done, took 5.636209ms
I260410 20:58:29.369388 112450 jobs/registry.go:1622  [T1,Vsystem,n1,job=1165834024648736769] 345  NEW SCHEMA CHANGE job 1165834024648736769: stepping through state succeeded
I260410 20:58:29.380157 112450 1@jobs/update.go:193  [T1,Vsystem,n1,job=1165834024648736769] 346 ={"Timestamp":1775854709380155732,"EventType":"status_change","JobID":1165834024648736769,"JobType":"NEW SCHEMA CHANGE","Description":"state changed to: succeeded","PreviousStatus":"running","NewStatus":"succeeded","RunNum":0}
I260410 20:58:29.387956 112329 jobs/wait.go:148  [T1,Vsystem,n1,client=127.0.0.1:49702,hostssl,user=root] 347  waited for 1 [1165834024648736769] queued jobs to complete 91.928909ms
    observer_test.go:124: 
        	Error Trace:	pkg/sql/catalog/lease/observer_test.go:124
        	Error:      	"1" is not greater than "1"
        	Test:       	TestLeaseObserver
        	Messages:   	new versions were not detected
W260410 20:58:29.388444 112590 15@kv/kvserver/intentresolver/intent_resolver.go:944  [-] 348  failed to gc transaction record: could not GC completed transaction anchored at /Table/70/1/1165834024648736769: node unavailable; try another peer
I260410 20:58:29.388594 112358 4@util/log/event_log.go:90  [T1,Vsystem,n1,client=127.0.0.1:49702,hostssl,user=root] 349 ={"Timestamp":1775854709388592371,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:49702","SessionID":"18a51a72fde52a3d0000000000000001","Duration":132810422}
I260410 20:58:29.388702 112261 sql/stats/automatic_stats.go:766  [T1,Vsystem,n1] 350  quiescing auto stats refresher
I260410 20:58:29.388723 112262 sql/stats/automatic_stats.go:804  [T1,Vsystem,n1] 351  quiescing auto stats misestimates refresher
W260410 20:58:29.388708 111949 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T1,Vsystem,n1,job=AUTO SPAN CONFIG RECONCILIATION id=1165834023952384001,rangefeed=sql-watcher-descriptor-rangefeed,mux_n=1,gen=2] 352  error closing mux rangefeed client: context canceled
W260410 20:58:29.388822 112340 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T1,Vsystem,n1,rangefeed=tenant-boundaries-watcher,mux_n=1,gen=2] 353  error closing mux rangefeed client: context canceled
W260410 20:58:29.388897 111793 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T1,Vsystem,n1,rangefeed=system-config-cache,mux_n=1,gen=2] 354  error closing mux rangefeed client: context canceled
W260410 20:58:29.388941 110891 sql/sqlliveness/slinstance/slinstance.go:337  [T1,Vsystem,n1] 355  exiting heartbeat loop
W260410 20:58:29.388951 110891 sql/sqlliveness/slinstance/slinstance.go:324  [T1,Vsystem,n1] 356  exiting heartbeat loop with error: node unavailable; try another peer
E260410 20:58:29.388964 110891 server/server_sql.go:535  [T1,Vsystem,n1] 357  failed to run update of instance with new session ID: node unavailable; try another peer
I260410 20:58:29.388992 110595 server/start_listen.go:113  [T1,Vsystem,n1] 358  server shutting down: instructing cmux to stop accepting
W260410 20:58:29.389058 112276 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T1,Vsystem,n1,rangefeed=table-stats-cache,mux_n=1,gen=2] 359  error closing mux rangefeed client: context canceled
I260410 20:58:29.389108 112263 sql/stats/automatic_stats.go:857  [T1,Vsystem,n1] 360  quiescing stats garbage collector
W260410 20:58:29.389026 111135 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T1,Vsystem,n1,rangefeed=lease,mux_n=1,gen=2] 361  error closing mux rangefeed client: context canceled
W260410 20:58:29.389211 111141 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T1,Vsystem,n1,rangefeed=sql_instances,mux_n=1,gen=2] 362  error closing mux rangefeed client: context canceled
W260410 20:58:29.389254 112396 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T1,Vsystem,n1,rangefeed=tenant-settings-watcher,mux_n=1,gen=2] 363  error closing mux rangefeed client: context canceled
W260410 20:58:29.389265 112436 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T1,Vsystem,n1,rangefeed=statement-hints-watcher,mux_n=1,gen=2] 364  error closing mux rangefeed client: context canceled
W260410 20:58:29.389293 111738 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T1,Vsystem,n1,rangefeed=settings-watcher,mux_n=1,gen=2] 365  error closing mux rangefeed client: context canceled
W260410 20:58:29.389328 111796 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T1,Vsystem,n1,rangefeed=spanconfig-subscriber,mux_n=1,gen=2] 366  error closing mux rangefeed client: context canceled
W260410 20:58:29.389356 112383 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T1,Vsystem,n1,rangefeed=tenant-entry-watcher,mux_n=1,gen=2] 367  error closing mux rangefeed client: context canceled
W260410 20:58:29.389361 112020 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T1,Vsystem,n1,job=AUTO SPAN CONFIG RECONCILIATION id=1165834023952384001,rangefeed=sql-watcher-protected-ts-records-rangefeed,mux_n=1,gen=2] 368  error closing mux rangefeed client: context canceled
W260410 20:58:29.389458 112023 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T1,Vsystem,n1,job=AUTO SPAN CONFIG RECONCILIATION id=1165834023952384001,rangefeed=sql-watcher-zones-rangefeed,mux_n=1,gen=2] 369  error closing mux rangefeed client: context canceled
I260410 20:58:29.389519 111573 jobs/registry.go:1622  [T1,Vsystem,n1,job=1165834023952384001] 370  AUTO SPAN CONFIG RECONCILIATION job 1165834023952384001: stepping through state succeeded
W260410 20:58:29.394165 110099 server/server_sql.go:1846  [T1,Vsystem,n1] 371  server shutdown without a prior graceful drain
I260410 20:58:29.394231 110099 server/server_controller.go:416  [T1,Vsystem,n1] 372  server controller shutting down
I260410 20:58:29.394245 110099 server/server_controller.go:425  [T1,Vsystem,n1] 373  waiting for tenant servers to report stopped
--- FAIL: TestLeaseObserver (0.41s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=1</code>
 - <code>shard=3</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLeaseObserver.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-62823

sql/catalog/lease.TestLeaseObserver [failed](https://mesolite.cluster.engflow.com/invocations/default/924e164d-5d42-4b89-8cb0-03018e131011?testReportRun=4&testReportShard=3&testReportAttempt=1#targets-Ly9wa2cvc3FsL2NhdGFsb2cvbGVhc2U6bGVhc2VfdGVzdA==) on master @ [a90d34c60ef8b6d5f8a072185d7077b095c5d1dd](https://github.com/cockroachdb/cockroach/commits/a90d34c60ef8b6d5f8a072185d7077b095c5d1dd):


```
I260411 04:18:29.608650 91686 sql/stats/automatic_stats.go:857  [T10,Vtest-tenant,n1] 556  quiescing stats garbage collector
W260411 04:18:29.608643 91200 kv/kvclient/kvtenant/setting_overrides.go:65  [T10,Vtest-tenant,n1,tenant-connector] 557  error consuming TenantSettings RPC: grpc: context canceled [code 1/Canceled]
I260411 04:18:29.608674 90761 server/server_controller_channel_orchestrator.go:443  [n1,tenant-orchestration,tenant=test-tenant] 558  tenant "test-tenant" finishing their own control loop
I260411 04:18:29.608723 91722 4@util/log/event_log.go:90  [T10,Vtest-tenant,n1,client=127.0.0.1:35294,hostssl,user=root] 559 ={"Timestamp":1775881109608720938,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:35294","SessionID":"18a53275c6d65e1d0000000000000001","Duration":112686156}
W260411 04:18:29.608789 91199 kv/kvclient/kvtenant/connector.go:397  [T10,Vtest-tenant,n1,tenant-connector] 560  error consuming GossipSubscription RPC: grpc: context canceled [code 1/Canceled]
W260411 04:18:29.608839 91089 sql/sqlliveness/slinstance/slinstance.go:337  [T10,Vtest-tenant,n1] 561  exiting heartbeat loop
W260411 04:18:29.608814 91595 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T10,Vtest-tenant,n1,job=AUTO SPAN CONFIG RECONCILIATION id=1165920532392443905,rangefeed=sql-watcher-protected-ts-records-rangefeed,mux_n=1,gen=2] 562  error closing mux rangefeed client: context canceled
W260411 04:18:29.608902 89988 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T1,Vsystem,n1,rangefeed=system-config-cache,mux_n=1,gen=2] 564  error closing mux rangefeed client: context canceled
W260411 04:18:29.608976 91265 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T10,Vtest-tenant,n1,rangefeed=sql_instances,mux_n=1,gen=2] 565  error closing mux rangefeed client: context canceled
W260411 04:18:29.609020 89446 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T1,Vsystem,n1,rangefeed=sql_instances,mux_n=1,gen=2] 566  error closing mux rangefeed client: context canceled
W260411 04:18:29.609104 91810 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T10,Vtest-tenant,n1,rangefeed=statement-hints-watcher,mux_n=1,gen=2] 567  error closing mux rangefeed client: context canceled
W260411 04:18:29.609182 90311 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T1,Vsystem,n1,job=AUTO SPAN CONFIG RECONCILIATION id=1165920531260538881,rangefeed=sql-watcher-descriptor-rangefeed,mux_n=1,gen=2] 571  error closing mux rangefeed client: context canceled
W260411 04:18:29.609207 91589 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T10,Vtest-tenant,n1,job=AUTO SPAN CONFIG RECONCILIATION id=1165920532392443905,rangefeed=sql-watcher-descriptor-rangefeed,mux_n=1,gen=2] 573  error closing mux rangefeed client: context canceled
W260411 04:18:29.609265 90303 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T1,Vsystem,n1,job=AUTO SPAN CONFIG RECONCILIATION id=1165920531260538881,rangefeed=sql-watcher-protected-ts-records-rangefeed,mux_n=1,gen=2] 574  error closing mux rangefeed client: context canceled
W260411 04:18:29.609282 90716 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T1,Vsystem,n1,rangefeed=statement-hints-watcher,mux_n=1,gen=2] 575  error closing mux rangefeed client: context canceled
W260411 04:18:29.609340 90481 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T1,Vsystem,n1,rangefeed=table-stats-cache,mux_n=1,gen=2] 577  error closing mux rangefeed client: context canceled
W260411 04:18:29.609351 91458 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T10,Vtest-tenant,n1,rangefeed=system-config-cache,mux_n=1,gen=2] 578  error closing mux rangefeed client: context canceled
W260411 04:18:29.609364 91310 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T1,Vsystem,n1,rangefeed=settings-watcher,mux_n=1,gen=2] 579  error closing mux rangefeed client: context canceled
W260411 04:18:29.609416 91354 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T1,Vsystem,n1,rangefeed=tenant-entry-watcher,mux_n=1,gen=2] 581  error closing mux rangefeed client: context canceled
W260411 04:18:29.609427 91649 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T10,Vtest-tenant,n1,rangefeed=table-stats-cache,mux_n=1,gen=2] 582  error closing mux rangefeed client: context canceled
W260411 04:18:29.609451 90322 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T1,Vsystem,n1,job=AUTO SPAN CONFIG RECONCILIATION id=1165920531260538881,rangefeed=sql-watcher-zones-rangefeed,mux_n=1,gen=2] 583  error closing mux rangefeed client: context canceled
W260411 04:18:29.609496 91256 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T10,Vtest-tenant,n1,rangefeed=lease,mux_n=1,gen=2] 585  error closing mux rangefeed client: context canceled
W260411 04:18:29.608856 91089 sql/sqlliveness/slinstance/slinstance.go:324  [T10,Vtest-tenant,n1] 563  exiting heartbeat loop with error: node unavailable; try another peer
E260411 04:18:29.609533 91089 server/server_sql.go:535  [T10,Vtest-tenant,n1] 587  failed to run update of instance with new session ID: node unavailable; try another peer
I260411 04:18:29.609129 90511 sql/stats/automatic_stats.go:804  [T1,Vsystem,n1] 569  quiescing auto stats misestimates refresher
I260411 04:18:29.609127 90510 sql/stats/automatic_stats.go:766  [T1,Vsystem,n1] 570  quiescing auto stats refresher
W260411 04:18:29.609395 89368 sql/sqlliveness/slinstance/slinstance.go:337  [T1,Vsystem,n1] 580  exiting heartbeat loop
I260411 04:18:29.609586 91684 sql/stats/automatic_stats.go:766  [T10,Vtest-tenant,n1] 590  quiescing auto stats refresher
W260411 04:18:29.609563 89430 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T1,Vsystem,n1,rangefeed=lease,mux_n=1,gen=2] 588  error closing mux rangefeed client: context canceled
W260411 04:18:29.609283 91592 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T10,Vtest-tenant,n1,job=AUTO SPAN CONFIG RECONCILIATION id=1165920532392443905,rangefeed=sql-watcher-zones-rangefeed,mux_n=1,gen=2] 576  error closing mux rangefeed client: context canceled
W260411 04:18:29.609586 89368 sql/sqlliveness/slinstance/slinstance.go:324  [T1,Vsystem,n1] 591  exiting heartbeat loop with error: node unavailable; try another peer
E260411 04:18:29.609653 89368 server/server_sql.go:535  [T1,Vsystem,n1] 594  failed to run update of instance with new session ID: node unavailable; try another peer
W260411 04:18:29.609565 91472 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T10,Vtest-tenant,n1,rangefeed=settings-watcher,mux_n=1,gen=2] 589  error closing mux rangefeed client: context canceled
I260411 04:18:29.609507 91685 sql/stats/automatic_stats.go:804  [T10,Vtest-tenant,n1] 586  quiescing auto stats misestimates refresher
I260411 04:18:29.609485 89871 jobs/registry.go:1622  [T1,Vsystem,n1,job=1165920531260538881] 584  AUTO SPAN CONFIG RECONCILIATION job 1165920531260538881: stepping through state succeeded
W260411 04:18:29.609203 91529 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T1,Vsystem,n1,rangefeed=tenant-settings-watcher,mux_n=1,gen=2] 572  error closing mux rangefeed client: context canceled
W260411 04:18:29.609728 90591 rpc/drpc.go:408  [-] 595  drpc server error context canceled
W260411 04:18:29.609108 90036 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T1,Vsystem,n1,rangefeed=spanconfig-subscriber,mux_n=1,gen=2] 568  error closing mux rangefeed client: context canceled
W260411 04:18:29.609780 90553 rpc/drpc.go:408  [-] 596  drpc server error context canceled
W260411 04:18:29.609638 90630 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T1,Vsystem,n1,rangefeed=tenant-boundaries-watcher,mux_n=1,gen=2] 593  error closing mux rangefeed client: context canceled
I260411 04:18:29.609625 91372 jobs/registry.go:1622  [T10,Vtest-tenant,n1,job=1165920532392443905] 592  AUTO SPAN CONFIG RECONCILIATION job 1165920532392443905: stepping through state succeeded
I260411 04:18:29.613616 90762 1@util/log/event_log.go:90  [n1,tenant-orchestration,tenant=test-tenant] 597 ={"Timestamp":1775881109613614216,"EventType":"tenant_shared_service_stop","NodeID":1,"TenantID":10,"InstanceID":1,"TenantName":"test-tenant"}
W260411 04:18:29.613645 90762 sql/event_log.go:620  [n1,tenant-orchestration,tenant=test-tenant] 598  failed to start task to save 1 events in eventlog: node unavailable; try another peer
W260411 04:18:29.613666 90762 server/server_sql.go:1846  [T10,Vtest-tenant,n1] 599  server shutdown without a prior graceful drain
I260411 04:18:29.613789 90762 server/server_controller_channel_orchestrator.go:487  [n1,tenant-orchestration,tenant=test-tenant] 600  propagate-ungraceful-stop task terminating
I260411 04:18:29.619031 90760 server/server_controller_channel_orchestrator.go:288  [T1,Vsystem,n1] 601  propagate-close task terminating
W260411 04:18:29.619333 88402 server/server_sql.go:1846  [T1,Vsystem,n1] 602  server shutdown without a prior graceful drain
I260411 04:18:29.619477 88402 server/server_controller.go:416  [T1,Vsystem,n1] 603  server controller shutting down
I260411 04:18:29.619571 88402 server/server_controller.go:425  [T1,Vsystem,n1] 604  waiting for tenant servers to report stopped
--- FAIL: TestLeaseObserver (0.71s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=4</code>
 - <code>shard=3</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLeaseObserver.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

sql/catalog/lease.TestLeaseObserver [failed](https://mesolite.cluster.engflow.com/invocations/default/bdb9d82d-4de6-40b1-afbb-353e31dee0dc?testReportRun=10&testReportShard=3&testReportAttempt=1#targets-Ly9wa2cvc3FsL2NhdGFsb2cvbGVhc2U6bGVhc2VfdGVzdA==) on master @ [f9bcbc4fdf0ba5236f6312c00becf303c29f98f9](https://github.com/cockroachdb/cockroach/commits/f9bcbc4fdf0ba5236f6312c00becf303c29f98f9):


```
I260412 04:16:05.026871 109917 sql/schemachanger/scrun/scrun.go:186  [T1,Vsystem,n1,job=NEW SCHEMA CHANGE id=1166203174861602817] 313  executing declarative schema change PostCommitPhase stage 1 of 1 with 3 MutationType ops (rollback=false) for ALTER TABLE
I260412 04:16:05.045197 109917 sql/internal.go:1995  [T1,Vsystem,n1,job=NEW SCHEMA CHANGE id=1166203174861602817] 314  waiting for 1 modified and 0 deleted descriptors
I260412 04:16:05.047147 109952 sql/catalog/lease/descriptor_state.go:198  [n1] 315  new lease: tabledesc.immutable: {ID: 104, Version: 3, ModificationTime: "1775967365.026737361,0", ParentID: 100, ParentSchemaID: 101, State: PUBLIC, NextColumnID: 3, Columns: [{ID: 1, TypeID: 20, Null: false}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2]}], Mutations: [{MutationID: 1, Direction: ADD, State: WRITE_ONLY, Column: {ID: 2, TypeID: 20, Null: true, State: ADD, MutationID: 1}}], PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}], StoreColumns: [2]}]}
I260412 04:16:05.047630 109917 sql/catalog/lease/lease.go:619  [T1,Vsystem,n1,job=NEW SCHEMA CHANGE id=1166203174861602817] 316  waiting for 1 leases to expire: desc=[{t 104 2}]
I260412 04:16:05.051265 109917 sql/catalog/lease/lease.go:613  [T1,Vsystem,n1,job=NEW SCHEMA CHANGE id=1166203174861602817] 317  all leases have expired at 1775967365.049330617,0: desc=[{t 104 2}]
I260412 04:16:05.051439 109917 sql/internal.go:2025  [T1,Vsystem,n1,job=NEW SCHEMA CHANGE id=1166203174861602817] 318  waiting for descriptors done, took 6.243512ms
I260412 04:16:05.051697 109917 sql/schemachanger/scrun/scrun.go:186  [T1,Vsystem,n1,job=NEW SCHEMA CHANGE id=1166203174861602817] 319  executing declarative schema change PostCommitNonRevertiblePhase stage 1 of 1 with 4 MutationType ops (rollback=false) for ALTER TABLE
I260412 04:16:05.069126 109917 5@util/log/event_log.go:90  [T1,Vsystem,n1,job=NEW SCHEMA CHANGE id=1166203174861602817] 320 ={"Timestamp":1775967365051579897,"EventType":"finish_schema_change","InstanceID":1,"DescriptorID":104,"LatencyNanos":60352339}
I260412 04:16:05.069191 109917 sql/internal.go:1995  [T1,Vsystem,n1,job=NEW SCHEMA CHANGE id=1166203174861602817] 321  waiting for 1 modified and 0 deleted descriptors
I260412 04:16:05.071020 110037 sql/catalog/lease/descriptor_state.go:198  [n1] 322  new lease: tabledesc.immutable: {ID: 104, Version: 4, ModificationTime: "1775967365.051579897,0", ParentID: 100, ParentSchemaID: 101, State: PUBLIC, NextColumnID: 3, Columns: [{ID: 1, TypeID: 20, Null: false}, {ID: 2, TypeID: 20, Null: true}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2]}], PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}], StoreColumns: [2]}]}
I260412 04:16:05.071512 109917 sql/catalog/lease/lease.go:619  [T1,Vsystem,n1,job=NEW SCHEMA CHANGE id=1166203174861602817] 323  waiting for 1 leases to expire: desc=[{t 104 3}]
I260412 04:16:05.074970 109917 sql/catalog/lease/lease.go:613  [T1,Vsystem,n1,job=NEW SCHEMA CHANGE id=1166203174861602817] 324  all leases have expired at 1775967365.073493118,0: desc=[{t 104 3}]
I260412 04:16:05.075193 109917 sql/internal.go:2025  [T1,Vsystem,n1,job=NEW SCHEMA CHANGE id=1166203174861602817] 325  waiting for descriptors done, took 6.001815ms
I260412 04:16:05.075480 109917 jobs/registry.go:1622  [T1,Vsystem,n1,job=1166203174861602817] 326  NEW SCHEMA CHANGE job 1166203174861602817: stepping through state succeeded
I260412 04:16:05.087167 109917 1@jobs/update.go:193  [T1,Vsystem,n1,job=1166203174861602817] 327 ={"Timestamp":1775967365087164184,"EventType":"status_change","JobID":1166203174861602817,"JobType":"NEW SCHEMA CHANGE","Description":"state changed to: succeeded","PreviousStatus":"running","NewStatus":"succeeded","RunNum":0}
I260412 04:16:05.095244 109846 jobs/wait.go:148  [T1,Vsystem,n1,client=127.0.0.1:34888,hostssl,user=root] 328  waited for 1 [1166203174861602817] queued jobs to complete 100.321881ms
    observer_test.go:124: 
        	Error Trace:	pkg/sql/catalog/lease/observer_test.go:124
        	Error:      	"1" is not greater than "1"
        	Test:       	TestLeaseObserver
        	Messages:   	new versions were not detected
W260412 04:16:05.095758 109576 15@kv/kvserver/intentresolver/intent_resolver.go:944  [-] 329  failed to gc transaction record: could not GC completed transaction anchored at /Table/15/5/"UPDATE TABLE METADATA CACHE"/105/0: node unavailable; try another peer
I260412 04:16:05.095837 108026 server/start_listen.go:113  [T1,Vsystem,n1] 330  server shutting down: instructing cmux to stop accepting
W260412 04:16:05.096025 109840 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T1,Vsystem,n1,rangefeed=tenant-entry-watcher,mux_n=1,gen=2] 333  error closing mux rangefeed client: context canceled
W260412 04:16:05.096063 109491 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T1,Vsystem,n1,job=AUTO SPAN CONFIG RECONCILIATION id=1166203173990432769,rangefeed=sql-watcher-descriptor-rangefeed,mux_n=1,gen=2] 334  error closing mux rangefeed client: context canceled
W260412 04:16:05.096167 109192 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T1,Vsystem,n1,rangefeed=system-config-cache,mux_n=1,gen=2] 335  error closing mux rangefeed client: context canceled
I260412 04:16:05.095887 109681 sql/stats/automatic_stats.go:766  [T1,Vsystem,n1] 331  quiescing auto stats refresher
I260412 04:16:05.095902 109714 sql/stats/automatic_stats.go:804  [T1,Vsystem,n1] 332  quiescing auto stats misestimates refresher
W260412 04:16:05.096310 108974 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T1,Vsystem,n1,rangefeed=settings-watcher,mux_n=1,gen=2] 336  error closing mux rangefeed client: context canceled
W260412 04:16:05.096349 109904 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T1,Vsystem,n1,rangefeed=tenant-settings-watcher,mux_n=1,gen=2] 337  error closing mux rangefeed client: context canceled
I260412 04:16:05.096139 109715 sql/stats/automatic_stats.go:857  [T1,Vsystem,n1] 338  quiescing stats garbage collector
W260412 04:16:05.096441 109663 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T1,Vsystem,n1,rangefeed=table-stats-cache,mux_n=1,gen=2] 339  error closing mux rangefeed client: context canceled
I260412 04:16:05.096461 109802 4@util/log/event_log.go:90  [T1,Vsystem,n1,client=127.0.0.1:34888,hostssl,user=root] 340 ={"Timestamp":1775967365096459015,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:34888","SessionID":"18a580e8b0fc542b0000000000000001","Duration":138827966}
W260412 04:16:05.096501 109876 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T1,Vsystem,n1,rangefeed=statement-hints-watcher,mux_n=1,gen=2] 341  error closing mux rangefeed client: context canceled
W260412 04:16:05.096534 109744 rpc/drpc.go:408  [-] 342  drpc server error context canceled
W260412 04:16:05.096564 109711 rpc/drpc.go:408  [-] 343  drpc server error context canceled
W260412 04:16:05.096556 109440 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T1,Vsystem,n1,job=AUTO SPAN CONFIG RECONCILIATION id=1166203173990432769,rangefeed=sql-watcher-protected-ts-records-rangefeed,mux_n=1,gen=2] 344  error closing mux rangefeed client: context canceled
W260412 04:16:05.096579 108481 sql/sqlliveness/slinstance/slinstance.go:337  [T1,Vsystem,n1] 345  exiting heartbeat loop
W260412 04:16:05.096613 109761 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T1,Vsystem,n1,rangefeed=tenant-boundaries-watcher,mux_n=1,gen=2] 346  error closing mux rangefeed client: context canceled
W260412 04:16:05.096620 108481 sql/sqlliveness/slinstance/slinstance.go:324  [T1,Vsystem,n1] 347  exiting heartbeat loop with error: node unavailable; try another peer
E260412 04:16:05.096637 108481 server/server_sql.go:535  [T1,Vsystem,n1] 348  failed to run update of instance with new session ID: node unavailable; try another peer
W260412 04:16:05.096664 108697 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T1,Vsystem,n1,rangefeed=sql_instances,mux_n=1,gen=2] 349  error closing mux rangefeed client: context canceled
W260412 04:16:05.096694 109033 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T1,Vsystem,n1,rangefeed=spanconfig-subscriber,mux_n=1,gen=2] 350  error closing mux rangefeed client: context canceled
W260412 04:16:05.096716 109392 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T1,Vsystem,n1,job=AUTO SPAN CONFIG RECONCILIATION id=1166203173990432769,rangefeed=sql-watcher-zones-rangefeed,mux_n=1,gen=2] 351  error closing mux rangefeed client: context canceled
W260412 04:16:05.096753 108621 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:417  [T1,Vsystem,n1,rangefeed=lease,mux_n=1,gen=2] 352  error closing mux rangefeed client: context canceled
I260412 04:16:05.096781 108968 jobs/registry.go:1622  [T1,Vsystem,n1,job=1166203173990432769] 353  AUTO SPAN CONFIG RECONCILIATION job 1166203173990432769: stepping through state succeeded
W260412 04:16:05.101400 107570 server/server_sql.go:1846  [T1,Vsystem,n1] 354  server shutdown without a prior graceful drain
I260412 04:16:05.101474 107570 server/server_controller.go:416  [T1,Vsystem,n1] 355  server controller shutting down
I260412 04:16:05.101513 107570 server/server_controller.go:425  [T1,Vsystem,n1] 356  waiting for tenant servers to report stopped
--- FAIL: TestLeaseObserver (0.48s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=10</code>
 - <code>shard=3</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLeaseObserver.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/sql/catalog/lease/observer_test.go
+++ b/pkg/sql/catalog/lease/observer_test.go
@@ -13,6 +13,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/base"
 	"github.com/cockroachdb/cockroach/pkg/sql/catalog/descpb"
 	"github.com/cockroachdb/cockroach/pkg/sql/catalog/lease"
+	"github.com/cockroachdb/cockroach/pkg/testutils"
 	"github.com/cockroachdb/cockroach/pkg/testutils/serverutils"
 	"github.com/cockroachdb/cockroach/pkg/testutils/sqlutils"
 	"github.com/cockroachdb/cockroach/pkg/util/ctxgroup"
@@ -121,7 +122,12 @@ func TestLeaseObserver(t *testing.T) {
 	// Add the first column and validate all versions are observed.
 	sqlRunner.Exec(t, "ALTER TABLE t ADD COLUMN v INT")
 	// Validate the schema change was observed.
-	require.Greaterf(t, maxVersion.Load(), int64(1), "new versions were not detected")
+	testutils.SucceedsSoon(t, func() error {
+		if maxVersion.Load() <= int64(1) {
+			return errors.New("new versions were not detected")
+		}
+		return nil
+	})
 	// Confirm the observer stops firing after unregistering.
 	unregisterFn()
 	maybeCloseSchemaChangeComplete()

